packages/nns/src/governance.canister.ts

```diff
@@ -156,31 +159,114 @@ export class GovernanceCanister {
     includeEmptyNeurons?: boolean;
     includePublicNeurons?: boolean;
     neuronSubaccounts?: NeuronSubaccount[];
+  }): Promise<NeuronInfo[]> => {
+    const useOldMethod =
+      isNullish(includeEmptyNeurons) &&
+      isNullish(includePublicNeurons) &&
+      certified;
+
+    if (useOldMethod) {
+      return this.fetchNeuronsWithOldMethodAndNoPagination({
+        neuronIds,
+      });
+    }
+
+    // Fetch the first page to get the total number of pages
+    const { neurons: firstPageNeurons, totalPages } =
+      await this.fetchNeuronsPage({
+        certified,
+        neuronIds,
+        includeEmptyNeurons,
+        includePublicNeurons,
+        neuronSubaccounts,
+      });
+
+    // https://github.com/dfinity/ic/blob/de17b0d718d6f279e9da8cd0f1b5de17036a6102/rs/nns/governance/api/src/ic_nns_governance.pb.v1.rs#L3543
+    if (totalPages < 2n) return firstPageNeurons;
```

Comment:
it seems like based on the below way of iterating, you don't actually need this conditional